language-server/src/main/java/org/sonarlint/languageserver/SonarLintLanguageServer.java

```diff
@@ -531,8 +601,25 @@ public void didChangeWatchedFiles(DidChangeWatchedFilesParams params) {
     // No watched files
   }
 
-  public StandaloneSonarLintEngine getEngine() {
-    return engine;
+  // TODO this method never seems to get triggered...
+  // Users must restart language server after adding new workspace folder.
```

Comment:
Why?

Reply:
Without having the added folder at server side, the `basedir` won't be correct for the analysis. In that case we fall back to the parent dir of the file as the base dir.
(Btw the fallback to parent dir is new, the PR fixes an NPE there.)